pandas/tests/frame/test_indexing.py

```diff
@@ -233,67 +233,105 @@ def inc(x):
         expected = pd.DataFrame([[-1, inc], [inc, -1]])
         tm.assert_frame_equal(df, expected)
 
-    def test_getitem_boolean(self):
+    def test_getitem_boolean_from_datetime(self, datetime_frame):
         # boolean indexing
-        d = self.tsframe.index[10]
-        indexer = self.tsframe.index > d
+        d = datetime_frame.index[10]
+        indexer = datetime_frame.index > d
         indexer_obj = indexer.astype(object)
 
-        subindex = self.tsframe.index[indexer]
-        subframe = self.tsframe[indexer]
+        subindex = datetime_frame.index[indexer]
+        subframe = datetime_frame[indexer]
 
         tm.assert_index_equal(subindex, subframe.index)
         with pytest.raises(ValueError, match='Item wrong length'):
-            self.tsframe[indexer[:-1]]
+            datetime_frame[indexer[:-1]]
 
-        subframe_obj = self.tsframe[indexer_obj]
+        subframe_obj = datetime_frame[indexer_obj]
         assert_frame_equal(subframe_obj, subframe)
 
         with pytest.raises(ValueError, match='boolean values only'):
-            self.tsframe[self.tsframe]
+            datetime_frame[datetime_frame]
 
         # test that Series work
-        indexer_obj = Series(indexer_obj, self.tsframe.index)
+        indexer_obj = Series(indexer_obj, datetime_frame.index)
 
-        subframe_obj = self.tsframe[indexer_obj]
+        subframe_obj = datetime_frame[indexer_obj]
         assert_frame_equal(subframe_obj, subframe)
 
         # test that Series indexers reindex
         # we are producing a warning that since the passed boolean
         # key is not the same as the given index, we will reindex
         # not sure this is really necessary
         with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
-            indexer_obj = indexer_obj.reindex(self.tsframe.index[::-1])
-            subframe_obj = self.tsframe[indexer_obj]
+            indexer_obj = indexer_obj.reindex(datetime_frame.index[::-1])
+            subframe_obj = datetime_frame[indexer_obj]
             assert_frame_equal(subframe_obj, subframe)
 
         # test df[df > 0]
```

Comment:
this i not what i meant. you want a single test that is parametrized.

Reply:
Well, these fixtures can't be parametrized without some serious gymnastics that I'm not going to go into here (limitations on using fixtures as parameters, limitations on combining fixtures into a fixture with e.g. yield, having to paste fixtures together in conftest [not as easy as for DTYPES], importing fixtures directly).

In this PR, you can have a simple translation, or a simple split.

Reply:
let's revert this particular section (IOW leave the loop). this is getting too much / messy.